components/scream/src/share/grid/point_grid.cpp

```diff
@@ -121,6 +123,36 @@ create_point_grid (const std::string& grid_name,
 
   grid->set_dofs(dofs_gids);
 
+  using device_type       = DefaultDevice;
+  using kokkos_types      = KokkosTypes<device_type>;
+  using geo_view_type     = kokkos_types::view_1d<double>;
+  using KT                = KokkosTypes<DefaultDevice>;
+  using C                 = scream::physics::Constants<double>;
+
+  // Store cell area, longitude, and latitude in geometry data.
+  // For  longitude and latitude, set values to NaN since they
+  // are currently not required from any application using PointGrid
+  geo_view_type area("area", num_my_cols);
+  geo_view_type lon ("lon",  num_my_cols);
+  geo_view_type lat ("lat",  num_my_cols);
+
+  // Estimate cell area for a uniform grid by taking the surface area
+  // of the earth divided by the number of columns
+  const double rearth    = C::r_earth;
+  const double pi        = C::Pi;
+  const double cell_area = 4*pi*rearth*rearth/num_my_cols;
+
+  const auto policy = ekat::ExeSpaceUtils<KT::ExeSpace>::get_default_team_policy(num_my_cols, num_vertical_lev);
+  Kokkos::parallel_for("area_loop", policy, KOKKOS_LAMBDA (const KT::MemberType& team) {
+    const int i = team.league_rank();
+    area(i) = cell_area;
+    lon(i) = std::nan("");
+    lat(i) = std::nan("");
```

Comment:
Am I understanding correctly that we are adding lat and lon to geo_view object now because we know it will be useful, but we are punting on actually giving it values at this point, assigning these things to NaN instead? Or is lat and lon getting assigned somewhere else?

Reply:
This function has no info about the sphere, so it can't setup lat/lon. It would require a massive amount of code to figure out the lat/lon coords of N uniformly distributed points on the sphere... On the other hand, the DynamicsDrivenGridsManager can extract lat/lon info from Homme, and will set them in the geo_data of the phys grid.

So we will lack lat/lon info for standalone shoc tests, but we will have that info for actually coupled (i.e., with dynamics) runs.

Reply:
Note: rrtmgp might actually need lat/lon info (for day/night/angle stuff), I am not sure. If so, in order to allow small standalone tests, we will have to either a) find a way to build a small PointGrid that _does_ have lat/lon info, or b) simply load small grids from file (with lat/lon info stored).

Reply:
I think that when the dynamics grid is used it will have lat defined, but we don't need it for the PointGrid and so we just set to NaN so that if someone tries to use it they will get an exception that shows it isn't set.